lib/cipherstash/protect/model/crud.rb

```diff
@@ -55,6 +55,30 @@ def protect_map_attributes(records)
               attributes
             end
           end
+
+          def match(arg = {})
+            unless arg.size == 1
+              raise CipherStash::Protect::Error, "Unable to execute text match query. Incorrect args passed. Example usage: model.match(email: 'test')"
+            end
+
+            unless arg.values.first.instance_of?(String)
+              raise CipherStash::Protect::Error, "Value passed to match query must be of type String. Got #{arg.values.first.inspect()}."
+            end
+
+            virt_attr = arg.keys.first
+            searchable_text_attr = protect_search_attrs[virt_attr]&.fetch(:searchable_text_attribute, nil)&.keys&.first
```

Comment:
I don't think I've seen this syntax - what is the `&` doing here?

Reply:
It's Ruby's version of the elvis operator (`foo?.bar?.baz`); equivalent to a big `foo && foo.bar && foo.bar.baz` chain.